lib/http2.c

```diff
@@ -1720,17 +1731,7 @@ static ssize_t http2_recv(struct Curl_easy *data, int sockindex,
 
       stream->pausedata = NULL;
       stream->pauselen = 0;
-
-      /* When NGHTTP2_ERR_PAUSE is returned from
-         data_source_read_callback, we might not process DATA frame
-         fully.  Calling nghttp2_session_mem_recv() again will
-         continue to process DATA frame, but if there is no incoming
-         frames, then we have to call it again with 0-length data.
-         Without this, on_stream_close callback will not be called,
-         and stream could be hanged. */
-      if(h2_process_pending_input(data, httpc, err) != 0) {
-        return -1;
-      }
+      H2BUGF(infof(data, "[SUPER-6847] do not process pending input data when copying pausedata to the caller"));
```

Comment:
Removing this process prevents the next frame to come in to the same stream buffer before handling paused data. 
https://github.com/curl/curl/pull/10529
> do not process pending input data when copying pausedata to the caller